crowsnest/v4l2/ctl.py

```
@@ -0,0 +1,262 @@
+    """
+    Get the available formats of a given device
+    """
+    try:
+        fd = os.open(device_path, os.O_RDWR)
+        fmt = raw.v4l2_fmtdesc()
+        frmsize = raw.v4l2_frmsizeenum()
+        frmival = raw.v4l2_frmivalenum()
+        fmt.index = 0
+        fmt.type = constants.V4L2_BUF_TYPE_VIDEO_CAPTURE
+        formats = {}
+        for fmt in utils.ioctl_iter(fd, raw.VIDIOC_ENUM_FMT, fmt):
```

Ruff (0.15.1)

[warning] Loop control variable `fmt` overrides iterable it iterates

(B020)